frontend/src/core/diff/components/TranslationDiff.js

```diff
@@ -0,0 +1,59 @@
+/* @flow */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import DiffMatchPatch from 'diff-match-patch';
+
+import './TranslationDiff.css';
+
+
+type InternalProps = {|
+    base: string,
+    target: string,
+|};
+
+
+const dmp = new DiffMatchPatch();
+
+
+/**
+ * Render diff between the base and the target string.
+ * 
+ * Unchanged slices are wrapped in <span>.
+ * Added slices are wrapped in <ins>.
+ * Removed slices are wrapped in <del>.
+ */
+export class TranslationDiffBase extends React.Component<InternalProps> {
+    getDiff(base: string, target: string) {
+        const diff = dmp.diff_main(base, target);
+    
+        dmp.diff_cleanupSemantic(diff);
+        dmp.diff_cleanupEfficiency(diff);
+    
+        return diff;
+    }    
+
+    render() {
+        const { base, target } = this.props;
+        const diff = this.getDiff(base, target)
+
+        return diff.map((item, index) => {
+            let type = item[0];
+            let slice = item[1];
+    
+            switch(type) {
+                case DiffMatchPatch.DIFF_INSERT:
+                    return <ins key={ index }>{ slice }</ins>;
+    
+                case DiffMatchPatch.DIFF_DELETE:
+                    return <del key={ index }>{ slice }</del>;
+    
+                default:
+                    return <span key={ index }>{ slice }</span>;
+            }
+        });    
+    }
+}
+
+
+export default connect()(TranslationDiffBase);
```

Comment:
This component doesn't need to be connected. We only "connect" components that need a direct access to the redux store, which is not the case here.